File: Library/DRandom.py
/*******************************************************************************
*  Copyright by the contributors to the Dafny Project
*  SPDX-License-Identifier: MIT
*******************************************************************************/

import random

class DRandom {

  def ctor__(self) {
  }

  def Coin(): {
    var r := random.getrandbits(1)
    return True
  }
}

File: Library/DRandom.go
/*******************************************************************************
*  Copyright by the contributors to the Dafny Project
*  SPDX-License-Identifier: MIT
*******************************************************************************/

package DafnyLibraries

import(
  "math/rand"
)

func *DRandom Ctor__(self) {
}

func *DRandom Coin() bool {
  return rand.Intn(2)
}


